sklbench/benchmarks/distr_sklearn_estimator.py

```diff
@@ -0,0 +1,312 @@
+# ===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import io
+import json
+import logging
+import os
+from importlib.metadata import PackageNotFoundError, version
+from typing import Dict, List, Union
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator
+from sklearn.metrics import (  # accuracy metrics; regression metrics; clustering metrics
+    accuracy_score,
+    balanced_accuracy_score,
+    completeness_score,
+    davies_bouldin_score,
+    homogeneity_score,
+    log_loss,
+    mean_squared_error,
+    r2_score,
+    roc_auc_score,
+)
+
+from ..datasets import load_data
+from ..datasets.transformer import split_and_transform_data
+from ..utils.bench_case import get_bench_case_value, get_data_name
+from ..utils.common import custom_format, get_module_members
+from ..utils.config import bench_case_filter
+from ..utils.custom_types import BenchCase, Numeric, NumpyNumeric
+from ..utils.logger import logger
+from ..utils.measurement import measure_case
+from ..utils.special_params import assign_case_special_values_on_run
+from .common import main_template
+from .sklearn_estimator import estimator_to_task, get_number_of_classes, get_estimator
+import dpctl
+import dpctl.tensor as dpt
+from mpi4py import MPI
+
+
+def get_estimator_methods(bench_case: BenchCase) -> Dict[str, List[str]]:
+    # default estimator methods
+    estimator_methods = get_bench_case_value(bench_case, f"algorithm:estimator_methods", {"training": "fit", "inference": "predict"})
+    for stage in estimator_methods.keys():
+        estimator_methods[stage] = estimator_methods[stage].split("|")
+    return estimator_methods
+
+
+def get_subset_metrics_of_estimator(

```

Comment:
Duplication of function

Reply:
There are some small differences - conversion from dpctl to numpy arrays, removal of some metrics not used in distributed bench (predict_proba, manifold, support vectors).

Generally this can either be cleaned up further to specifically only account for relevant metrics or imported from sklearn_estimator with array conversion done before + small revisions to original function. Any preference?